tex/polyglossia-latin.lua

```diff
@@ -0,0 +1,39 @@
+require('polyglossia-punct')
+
+-- For ecclesiastic Latin (and sometimes for Italian) a very small space is
+-- used for the punctuation. The ecclesiastic package uses a space of
+-- 0.3\fontdimen2, where \fontdimen2 is a interword space, which is typically
+-- between 1/4 and 1/3 of a quad. We choose a half of a \thinspace here.
+local hairspace = 0.08333 -- 1/12 quad
+
+local function activate_latin_punct()
+  polyglossia.activate_punct()
+  polyglossia.clear_spaced_characters()
+  polyglossia.add_left_spaced_character(':',hairspace)
+  polyglossia.add_left_spaced_character('!',hairspace)
+  polyglossia.add_left_spaced_character('?',hairspace)
+  polyglossia.add_left_spaced_character(';',hairspace)
+  polyglossia.add_left_spaced_character('‼',hairspace)
+  polyglossia.add_left_spaced_character('⁇',hairspace)
+  polyglossia.add_left_spaced_character('⁈',hairspace)
+  polyglossia.add_left_spaced_character('⁉',hairspace)
+  polyglossia.add_left_spaced_character('»',hairspace)
+  polyglossia.add_left_spaced_character('›',hairspace)
+  polyglossia.add_right_spaced_character('«',hairspace)
+  polyglossia.add_right_spaced_character('‹',hairspace)
+end
+
+-- So far, the following function is empty. It is provided for symmetry.
+local function deactivate_latin_punct()
```

Comment:
you should really use the babel french code, it allow to mix different spacing in one paragraph.

I really prefer to merge first babel french then your improvment

Reply:
The mixing of different spacings in one paragraph is also possible with the polyglossia code. You just have to say `xpg@punct=0` (or `xpg@frpt=0` with the current implementation). This happens for example when switching from French to German within a paragraph and the spacing really changes in this case. So there is no real problem.
The Lua code of babel may be superior in other respects, but it would take a long time for me to work in.
I would prefer to publish this solution first. That would help to get my Latin code work.

Reply:
@wehro the code is buggy but I agree it could be interesting to have a plan to swith to babel one step by step before merging your code. First by analysing if babel code will work with latin

Reply:
One aspect where the babel code seems to be superior is #52. The soul error with LuaTeX does not occur with babel (but then, the spacing of the guillements strikes me wrong with babel/LuaTeX)

Reply:
Having written that, I think merging @wehro's current work cannot harm. It certainly improves the situation for Latin (and probably sanskrit in a next step). I agree that transforming it to a more babel way could be done after the merge.